docs/src/writing_good_rules.md

```diff
@@ -175,6 +176,68 @@ end
 ```
 to define the rules.
 
+## Ensure your pullback can accept the right types
+As a rule the number of types you need to accept in a pullback is theoretically unlimitted, but practically highly constrained to inline with the primal return type.
+The three kinds of inputs you will practically need to accept one or more of: _natural tangents_, _structural tangents_, and _thunks_.
+You do not in general have to handle `AbstractZero`s as the AD system will not call the pullback if the input is a zero, since the output will also be.
+Some more background information on these types can be found in [the design notes](@ref manytypes).
+In many cases these all all tangents can be treated the same: tangent types overload a bunch of linear-operators, and the majority of functions used inside a pullback are linear operators.
+If you find linear operators from Base/stdlibs that are not supported, consider openning an issue or PR on the [ChainRulesCorejl repo](https://github.com/JuliaDiff/ChainRulesCore.jl/).
+
+### Natural tangents
```

Comment:
Not objecting to adding this, but don't we already say what the differences between natural and structural tangents are somewhere in the docs?

Reply:
In the design docs we probably do.
I am in-favour of having redundancy in the docs.
It makes them much more understandable, since noone reads them end-to-end.
Albeit, at the cost of making the docs harder to maintain.